### PR TITLE
[Elasticsearch] support for setting preconditions

### LINKED ISSUE #210: [Elasticsearch] Support for SearchPreCondition
| Q                     | A
| --------------------- | -----
| Bug report?           | no
| Feature request?      | yes
| BC Break report?      | no
| RFC?                  | no

Related to #182, the Elasticsearch integration doesn't support (_handle_) SearchPreCondition yet.

### PR DESCRIPTION
**Pull request by dkarlovi**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #210 
| License       | MIT
| Doc PR        | N/A?

### CODE REVIEW

**Comment by dkarlovi** [comment]
@sstok  most of the changes have been done by PHP CS Fixer. Shouldn't it be run across the whole codebase so these kinds of changes don't trickle down with each PR?

File: lib/Elasticsearch/Tests/QueryConditionGeneratorTest.php [hunk 1]
```diff
@@ -42,43 +60,46 @@ public function it_generates_a_structure_of_root_level_fields()
     {
         $condition = $this->createCondition()
             ->field('id')
-                ->addSimpleValue(2)
```
Comment (sstok) [inline lib/Elasticsearch/Tests/QueryConditionGeneratorTest.php]: These need to be reverted. Builder don't need to be on the same level.

Comment (dkarlovi) [reply lib/Elasticsearch/Tests/QueryConditionGeneratorTest.php]: Agree, but that makes using PHP CS Fixer impossible?

Comment (sstok) [reply lib/Elasticsearch/Tests/QueryConditionGeneratorTest.php]: I ran the php-cs-fixer locally (using the makefile, and updated Docker image) but there were no reported problems 👩‍🔬 are you using a different version?

Comment (dkarlovi) [reply lib/Elasticsearch/Tests/QueryConditionGeneratorTest.php]: I'm just running `make`, don't know anything about anything. :laughing:

Comment (dkarlovi) [reply lib/Elasticsearch/Tests/QueryConditionGeneratorTest.php]: Also, reverted the style changes.

Comment (sstok) [reply lib/Elasticsearch/Tests/QueryConditionGeneratorTest.php]: Can you apply the correct style on the new code? 😅 Also if there is a specific fixer that's causing this please let me know so I can have a look 😛

Comment (dkarlovi) [reply lib/Elasticsearch/Tests/QueryConditionGeneratorTest.php]: I thought I did, this was also done by CS Fixer. :see_no_evil: 

I got the original coding style (and this one) by just running `make`, it did the whole thing and came up with this. Haven't run anything other than that, so whatever the current setup does, that's what it did.

Comment (dkarlovi) [reply lib/Elasticsearch/Tests/QueryConditionGeneratorTest.php]: @sstok any idea what to do here? If I apply CS style with CS fixer, I'll get the stuff we reverted again. :(

Comment (sstok) [reply lib/Elasticsearch/Tests/QueryConditionGeneratorTest.php]: Disable the CS fixer for now 😛 I will look into it. Btw, isn't this caused by PHPStorm? As far as I know php-cs-fixer should leave this things alone.

Comment (dkarlovi) [reply lib/Elasticsearch/Tests/QueryConditionGeneratorTest.php]: Nope, I'm not running style changes from PhpStorm anymore, it's done by CS Fixer 100%.

If I disable CS Fixer, how do I apply correct CS here? :rofl:

Comment (sstok) [reply lib/Elasticsearch/Tests/QueryConditionGeneratorTest.php]: By hand 😄 if you still know how to do this :trollface: Endless comments on what needs to be done different 🤣

Comment (dkarlovi) [reply lib/Elasticsearch/Tests/QueryConditionGeneratorTest.php]: Of course, but I don't know what the actual CS for this project is, this is the trap of automation, once the robots let you down, you're left sitting in sand wondering where did it all go wrong. :laughing:

**Comment by dkarlovi** [comment]
@sstok this should be it? If not, can you specifically point out what needs changing?

**Comment by sstok** [comment]
Thank you @dkarlovi

**Comment by dkarlovi** [comment]
Thank you @sstok, sorry for the pings. :)

**Pull request merged**